commands/cli/parse.go

```diff
@@ -13,20 +12,12 @@ import (
 	u "github.com/ipfs/go-ipfs/util"
 )
 
-// ErrInvalidSubcmd signals when the parse error is not found
-var ErrInvalidSubcmd = errors.New("subcommand not found")
-
 // Parse parses the input commandline string (cmd, flags, and args).
 // returns the corresponding command Request object.
 func Parse(input []string, stdin *os.File, root *cmds.Command) (cmds.Request, *cmds.Command, []string, error) {
-	path, input, cmd := parsePath(input, root)
-	if len(path) == 0 {
-		return nil, nil, path, ErrInvalidSubcmd
-	}
-
-	opts, stringVals, err := parseOptions(input)
+	path, opts, stringVals, cmd, err := parseOpts(input, root)
 	if err != nil {
-		return nil, cmd, path, err
+		return nil, nil, path, err
```

Comment:
not really in the scope of this PR, but I personally think theres no reason to return anything but `nil` if we have an error.